### PR TITLE
Update schema

### PR DESCRIPTION
**Pull request by alannotnerd**

### CODE REVIEW

File: src/handlers/executors/contribution.ts [hunk 1]
```diff
@@ -5,34 +5,45 @@ import { convertToAnyChainAddress } from '../utils/address'
...
   event: { data },
   block: {
     timestamp,
     block: { header }
-  },
-  extrinsic
+  }
 }: SubstrateEvent) => {
-  const [paraId, vaultId, contributor, amount, referralCode] = JSON.parse(
+  const [paraId, vaultId, contributor, amount, _] = JSON.parse(
```
Comment (crystalynliu) [inline src/handlers/executors/contribution.ts]: why cannot delete `-` directly?

File: schema.graphql [hunk 2]
```diff
@@ -2,26 +2,28 @@ type Contribution @entity {
...
   totalAmount: String!
   contributionStrategy: String!
   cap: String!
   endBlock: Int!
-  trieIndex: Int!
+}
+
+type VaultsDissolve @entity {
+  id: ID!
+  vault: Vaults! @index
```
Comment (crystalynliu) [inline schema.graphql]: If we have the same vaultId for one project with differernt status, one is dissolved, one is started, how to figure out it?

File: src/handlers/executors/vault.ts [hunk 3]
```diff
@@ -21,7 +24,7 @@ export const handleVaultCreated = async ({
     contributionStrategy,
     cap,
     endBlock,
-    trieIndex
+    _
```
Comment (crystalynliu) [inline src/handlers/executors/vault.ts]: Seems here, why cannot delete it directly?

File: src/handlers/executors/vault.ts [hunk 4]
```diff
@@ -109,21 +133,40 @@ export const handleVaultPhaseUpdated = async ({
...
   try {
-    await Vaults.remove(vault)
+    await VaultsDissolve.create({
+      id: idx.toString(),
+      vaultId: aggregateVaultId,
+      dissolvedBlockHeight: header.number.toNumber()
+    }).save()
+
+    let vault = await Vaults.get(aggregateVaultId)
+    vault.phase = 'Cancelled'
```
Comment (crystalynliu) [inline src/handlers/executors/vault.ts]: Maybe change to `Dissolved`?

File: src/handlers/executors/vault.ts [hunk 5]
```diff
@@ -33,43 +36,54 @@ export const handleVaultCreated = async ({
...
+    paraId.toString(),
+    vaultId[0].toString(),
+    vaultId[1].toString()
+  )
   let vaultRecord = await Vaults.get(vault)
   if (vaultRecord) {
     vaultRecord.contributionStrategy = contributionStrategy
-    vaultRecord.cap = ensureStrNumber(cap),
-    vaultRecord.endBlock = endBlock
+    ;(vaultRecord.cap = ensureStrNumber(cap)), (vaultRecord.endBlock = endBlock)
```
Comment (crystalynliu) [inline src/handlers/executors/vault.ts]: what's the `;` for here?

File: src/handlers/executors/vault.ts (hunk 4 above)
Comment (alannotnerd) [reply src/handlers/executors/vault.ts]: It's better to make consist with the enum in pallet.

File: schema.graphql [hunk 6]
```diff
@@ -1,27 +1,30 @@
...
   phase: String!
   contributions: Int!
   totalAmount: String!
   contributionStrategy: String!
   cap: String!
   endBlock: Int!
-  trieIndex: Int!
+}
+
+type DissolvedVault @entity {
```
Comment (crystalynliu) [inline schema.graphql:26]: Should we add totalAmount and contributions to dissolvedVault for debug easily in the future?

File: schema.graphql [hunk 7]
```diff
@@ -1,27 +1,32 @@
 type Contribution @entity {
   id: ID!
-
+  extrinsicHash: String! @index
   paraId: Int! @index
```
Comment (crystalynliu) [inline schema.graphql]: do we still need this paraId?

File: schema.graphql [hunk 8]
```diff
@@ -1,27 +1,32 @@
...
-  vaultId: String!
+  vault: Vaults! @index
   blockHeight: Int! @index
   account: String! @index
   amount: String!
-  referralCode: String
   timestamp: Date!
 }
 
 type Vaults @entity {
```
Comment (crystalynliu) [inline schema.graphql:11]: can we add firstSlot and lastSlot field in Vault schema?

**Review by crystalynliu — COMMENTED** [review]
Only two small question here

File: schema.graphql (hunk 8 above)
Comment (alannotnerd) [reply schema.graphql:11]: But the `id` includes that info.

Comment (crystalynliu) [reply schema.graphql:11]: I think it's a different meaning actually even we use it as vault id

Comment (alannotnerd) [reply schema.graphql:11]: Okay, got it. Should prompt the staging to production first?

**Review by crystalynliu — APPROVED** [review]

**Pull request merged**